File: chroma-league/src/main/java/net/booone/chromaleague/rest/playerlist/PlayerList.java
package net.booone.chromaleague.rest.playerlist;

import net.booone.chromaleague.state.RunningState;

import java.util.List;
import java.util.stream.Collectors;

public record PlayerList(List<Player> players) {

    public Player getActivePlayer() {
        return getByRiotId(RunningState.getGameState().getPlayerRiotId());
    }

    public List<String> getAllies() {
        return players.stream()
                .filter(p -> getActivePlayer().team() == p.team())
                .map(Player::riotIdGameName)
                .collect(Collectors.toList());
    }

    public List<String> getEnemies() {
        return players.stream()
                .filter(p -> getActivePlayer().team() != p.team())
                .map(Player::riotIdGameName)
                .collect(Collectors.toList());
    }

    public boolean isAlly(String riotIdGameName) {
        return getByRiotIdGameName(riotIdGameName).team() == getActivePlayer().team();
    }

    private Player getByRiotId(String riotId) {
        return players.stream()
                .filter(player -> player.riotId().equals(riotId))
                .findAny().orElseThrow(() -> new IllegalStateException("Couldn't find player with riotId " + riotId));
    }

    private Player getByRiotIdGameName(String riotIdGameName) {
        return players.stream()
                .filter(player -> player.riotIdGameName().equals(riotIdGameName))
                .findAny().orElseThrow(() -> new IllegalStateException("Couldn't find player with riotIdGameName " + riotIdGameName));
    }
}

File: src/rest/playerlist/PlayerList.ts
import {Player} from './Player.js';

export class PlayerList {
    readonly players: Player[];

    constructor(players: Player[]) {
        this.players = players;
    }

    getActivePlayer(playerRiotId: string): Player {
        return this.getByRiotId(playerRiotId);
    }

    getAllies(playerRiotId: string): string[] {
        const active = this.getActivePlayer(playerRiotId);
        return this.players
            .filter(p => active.team === p.team)
            .map(p => p.riotIdGameName);
    }

    getEnemies(playerRiotId: string): string[] {
        const active = this.getActivePlayer(playerRiotId);
        return this.players
            .filter(p => active.team !== p.team)
            .map(p => p.riotIdGameName);
    }

    isAlly(killerName: string, playerRiotId: string): boolean {
        const player = this.getByRiotIdGameName(killerName)
            // Probable bug in Riot API - sometimes returns ChampionName as KillerName
            ?? this.getByChampionName(killerName);
        if (!player) {
            throw new Error(`Couldn't find player with riotIdGameName ${killerName}`);
        }
        return player.team === this.getActivePlayer(playerRiotId).team;
    }

    private getByRiotId(riotId: string): Player {
        const player = this.players.find(p => p.riotId === riotId);
        if (!player) {
            throw new Error(`Couldn't find player with riotId ${riotId}`);
        }
        return player;
    }

    private getByRiotIdGameName(riotIdGameName: string): Player | undefined {
        return this.players.find(p => p.riotIdGameName === riotIdGameName);
    }

    private getByChampionName(championName: string): Player | undefined {
        return this.players.find(p => p.championName === championName);
    }
}
